src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -442,6 +442,7 @@ private boolean addConnection()
       // Speculative increment of totalConnections with expectation of success
       if (totalConnections.incrementAndGet() > config.getMaximumPoolSize()) {
          totalConnections.decrementAndGet(); // Pool is maxed out, so undo speculative increment of totalConnections
+         LOGGER.debug("{} - Cannot exceed maximum connections capacity: {}", poolName, config.getMaximumPoolSize());
```

Comment:
not tested.
I thought it is worth logging in scenario while answering question at stackoverflow s/he linked it to https://github.com/brettwooldridge/HikariCP/issues/109 !)